### PR TITLE
Fixed: Inventory Number display issue on Logged out state.

### PR DESCRIPTION
**Pull request by abulsayyad123**
Hi Team,
This is Fix for the Inventory number display issue in logged out state.
https://jira.crossroads.org.hk/browse/GCW-2153?focusedCommentId=31806&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-31806.

Please review the PR and let me know if any change is required.
Thanks.

### CODE REVIEW

File: app/controllers/item.js [hunk 1]
```diff
@@ -38,9 +37,10 @@ export default Ember.Controller.extend({
   }),
 
   hasDraftOrder: Ember.computed.alias("session.draftOrder"),
-  isOrderFulfilmentUser: Ember.computed(function() {
+  isOrderFulfilmentUser: Ember.computed('session.authToken', function() {
     let user = this.get('session.currentUser');
-    return user.hasRole('Order fulfilment');
+    let isLoggedIn = this.get('session.isLoggedIn');
+    return user && isLoggedIn ? user.hasRole('Order fulfilment') : false;
```
Comment (Darshan-Chauhan) [inline app/controllers/item.js]: I think we can improve the condition a bit.
User can't exist without logging in. So if User object is there that means he/she is logged in or is isLoggedIn is true that means User exists.

Comment (Darshan-Chauhan) [reply app/controllers/item.js]: Also `isOrderFulfilmentUser` computed property should only check is user has `Order fulfilment` role or not. It shouldn't check if User is logged in or not.

File: app/controllers/item.js [hunk 2]
```diff
@@ -38,9 +37,9 @@ export default Ember.Controller.extend({
   }),
 
   hasDraftOrder: Ember.computed.alias("session.draftOrder"),
-  isOrderFulfilmentUser: Ember.computed(function() {
+  isOrderFulfilmentUser: Ember.computed('session.authToken', function() {
```
Comment (Darshan-Chauhan) [inline app/controllers/item.js]: This comouted property should only get executed if user is available. As I've said before it should only check whether User has order fulfillment role or not. It shouldn't check if the user is logged in or not

**Comment by Darshan-Chauhan** [comment]
Also remove package lock file and add yarn.lock file

File: app/controllers/item.js [hunk 3]
```diff
@@ -38,9 +37,9 @@ export default Ember.Controller.extend({
...
     let user = this.get('session.currentUser');
-    return user.hasRole('Order fulfilment');
+    return user && user.hasRole('Order fulfilment');
```
Comment (Darshan-Chauhan) [inline app/controllers/item.js]: Unnecessary conditions

File: app/controllers/item.js (hunk 2 above)
Comment (abulsayyad123) [reply app/controllers/item.js]: If it is not executed after logged out, then 'Inventory number' remains visible on the screen.

File: app/controllers/item.js (hunk 3 above)
Comment (abulsayyad123) [reply app/controllers/item.js]: If this is executing after logout than this condition is required. Because it tries to fetch `roles` from `user`.

Comment (Darshan-Chauhan) [reply app/controllers/item.js]: Computed property shouldn't get called if no user exist

**Comment by Darshan-Chauhan** [comment]
@abulsayyad123 clear everything (current user and Everything) in session service's clear method.

**Pull request merged**